src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -31,48 +35,59 @@ public static void Square(ReadOnlySpan<uint> value, Span<uint> bits)
             // NOTE: useful thresholds needs some "empirical" testing,
             // which are smaller in DEBUG mode for testing purpose.
 
-            if (value.Length < SquareThreshold)
+            if (value.Length < MultiplyKaratsubaThreshold)
+                Naive(value, bits);
+            else if (value.Length < MultiplyToom3Threshold)
+                Karatsuba(value, bits);
+            else
+                Toom3(value, bits);
```

Comment:
nit: Style guidelines requires braces here: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

While a lot of BigInteger is older and predates our current guidelines, we do want new code to do the "right thing".